compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIViewComposeSceneLayer.uikit.kt

```diff
@@ -33,24 +33,53 @@ import androidx.compose.ui.window.ProvideContainerCompositionLocals
 import androidx.compose.ui.window.RenderingUIView
 import kotlin.coroutines.CoroutineContext
 import kotlinx.cinterop.CValue
+import kotlinx.cinterop.readValue
+import platform.CoreGraphics.CGPoint
+import platform.CoreGraphics.CGRectZero
 import platform.CoreGraphics.CGSize
+import platform.UIKit.NSLayoutConstraint
+import platform.UIKit.UIColor
+import platform.UIKit.UIEvent
 import platform.UIKit.UIView
 import platform.UIKit.UIViewControllerTransitionCoordinatorProtocol
 
 internal class UIViewComposeSceneLayer(
     private val composeContainer: ComposeContainer,
-    density: Density,
-    layoutDirection: LayoutDirection,
+    private val initDensity: Density,
+    private val initLayoutDirection: LayoutDirection,
     configuration: ComposeUIViewControllerConfiguration,
     focusStack: FocusStack<UIView>?,
     windowInfo: WindowInfo,
     compositionContext: CompositionContext,
     compositionLocalContext: CompositionLocalContext?,
 ) : ComposeSceneLayer {
 
+    override var focusable: Boolean = focusStack != null
+    private var onOutsidePointerEvent: ((dismissRequest: Boolean) -> Unit)? = null
+    private val rootView = composeContainer.view.window ?: composeContainer.view
+    private val backgroundView: UIView = object : UIView(
+        frame = CGRectZero.readValue()
+    ) {
+        init {
+            translatesAutoresizingMaskIntoConstraints = false
+            rootView.addSubview(this)
+            NSLayoutConstraint.activateConstraints(
+                getConstraintsToFillParent(this, rootView)
+            )
+        }
+
+        override fun pointInside(point: CValue<CGPoint>, withEvent: UIEvent?): Boolean {
+            //TODO pass invoke(true) on touch up event only when this touch event begins outside of layer bounds.
+            // Also it should be only one touch event (not multitouch with 2 and more touches).
+            // In other cases pass invoke(false)
+            onOutsidePointerEvent?.invoke(true)
```

Comment:
What about?:
```
val pointInside = super. pointInside(point, withEvent)
if (!pointInside) {
    onOutsidePointerEvent?.invoke(true)
}
return pointInside
```

Reply:
This is background view. So we always shoud call onOutsidePointerEvent. But in future we should add logic to handle different type of event like TOUCH_DOWN, TOUCH_UP.